packages/terra-framework-docs/src/terra-dev-site/test/pills/Pills/CollapsiblePills.test.jsx

```diff
@@ -0,0 +1,74 @@
+import React, { useRef, useState } from 'react';
+import classNames from 'classnames/bind';
+import Pills from 'terra-pills/lib/index';
+import styles from '../PillListTestCommon.module.scss';
+
+const cx = classNames.bind(styles);
+
+const CollapsibleBasicPills = () => {
+  const pillsData = [
+    {
+      label: 'asthma',
+      ref: useRef(),
```

Comment:
There are two similarly named APIs in React for creating ref objects, `createRef` and `useRef`.

[createRef](https://reactjs.org/docs/refs-and-the-dom.html#creating-refs) will do exactly as it says, create a new instance of a ref object and return it. Every time you call createRef, you get a new ref object instance.

[useRef](https://reactjs.org/docs/hooks-reference.html#useref), on the other hand, is a hook that will create a single instance of a ref object and return that instance every time it is called, which should be with every update of component it's used in.

```jsx
const myRef = useRef();

// is functionally equivalent to

const myRef = useState(createRef())[0];
```

That said, your usage of useRef here should likely be replaced with createRef. Though your usage doesn't technically violate the [rules of hooks](https://reactjs.org/docs/hooks-rules.html), it's only because you're creating this pillsData object with every update, unnecessarily. The value is only used the first render to initialize state and is ignored thereafter. If you used createRef here, you could initialize your state like this:
```js

```jsx
const [pillsState] = useState(() => ([{
  label: 'asthma',
  ref: createRef(),
  id: 'terra-pills-example-disclosure-removable-pill-asthma',
}, ...]));
```